### ISSUE TITLE
[Slash commands] User mention returning two different values

### ISSUE THREAD

**Issue by iamk3v** [AlexzanderFlores/WOKCommands#103]
My args log different results of a mention when I try on pc vs mobile.
Ran on pc it returns an actual guildMember Object, while on mobile it returns just a mention string.

You can see the output here:
https://sourceb.in/DIfmkmsnTo

Not sure if this is intentional or not, but I would love to see it at least return the same value.

All help is much appreciated

**Comment by Totto16** [comment]
I tested it and got the same result, this seems to have something todo with discords behavior, its kinda weird, but its fixable, a nice thing we have for slash Commands is the ApplicationCommandOptionType, we could try to use a correct Type for discord to know this is a user, maybe that could change that, otherwise we have to preprocess the arg into a GuildMember Object.

**Comment by iamk3v** [comment]
> I tested it and got the same result, this seems to have something todo with discords behavior, its kinda weird, but its fixable, a nice thing we have for slash Commands is the ApplicationCommandOptionType, we could try to use a correct Type for discord to know this is a user, maybe that could change that, otherwise we have to preprocess the arg into a GuildMember Object.

Yeah i thought about the same thing by adding a options parameter to the module.exports.. (without any luck)

```
module.exports = {
    slash: true,
    testOnly: true,
    name: 'ban',
    description: 'Bans a user, **MUST BE A MENTION**, specify number of days of messages to delete with days argument',
    minArgs: 1,
    expectedArgs: '<user> [reason] [days]',
    guildOnly: true,
    options: [{
        type: 6,
    }],
    permissions: ["BAN_MEMBERS"],
    ```
    
    Is there an implemented feature in WOKCommands to specify ApplicationCommandOptions?

**Comment by Totto16** [comment]
No unfortunately there isnt, but I also implemented slash commands via pull request, so for completeness I try to also add that option, and I'll test if it works that way...

**Comment by Totto16** [comment]
![grafik](https://user-images.githubusercontent.com/32566573/119159212-50799b00-ba57-11eb-86d3-d443472e4e26.png)
The problme is not discord, I just realised, while implementing the ApplicationCommandOptions, that we have a getMemberIfExists function, that checks, if one of the args is such a mention string, so in the Picture above you can see, that mobile and desktop discord has some different string, we only check for <@! instead of also for <!

**Comment by iamk3v** [comment]
> ![grafik](https://user-images.githubusercontent.com/32566573/119159212-50799b00-ba57-11eb-86d3-d443472e4e26.png)
> The problme is not discord, I just realised, while implementing the ApplicationCommandOptions, that we have a getMemberIfExists function, that checks, if one of the args is such a mention string, so in the Picture above you can see, that mobile and desktop discord has some different string, we only check for <@! instead of also for <!

Ah okay, so i assume you´ll push an update which checks for both to fix mobile mentions?
Will you announce if / when the update goes live?

**Comment by Totto16** [comment]
I will do that, yes, but I am not the author / a maintainer of this project, so I'll pull request it then

**Comment by Totto16** [comment]
I coded everything, that you need to make that, so: you now have avaiable a  options property in the commands file, you have to decalre and array of objects (for one argument an object is enough) it can have these attributes: type(1-9) or the string like "user" or "USER", then you cna declare another optional choices key, you can put there a choices array of strings or ints for choices ONLY for this types, then you have the whole boolean attribute, if you dont set this to true, you get the user/role/channel object with VERY redeuced information, like this one
```
{
    avatar: 'f5ac82821e9f34aaf3f3732aad630030',
    bot: true,
    discriminator: '0537',
    id: '759395211713082724',
    public_flags: 0,
    username: 'Bot'
  }
```
this information gets send with the slash interaction type and you get it by default, however if whole is true, you get the whole user object fetched by discord.js, keep in mind, it gets it from the caceh, but if not present it has to fetch it, so you have a speed disadvantage, I would suggest, to fetch it yourself if you need more information, beacuse then you can reply instantly with interaction.loading() -> fetch all data needed and then -> interaction.send(...) and your user doesn't think the bo is slow

**Comment by iamk3v** [comment]
Thank you! i assume this new array is written with something like this:?
```
module.exports = {
    slash: true,
    options: [{
        type: 6,
    }],
    name: "example",
    ...
```

Would it be possible to provide an exact example? Haven't seen an update on docs with the newly added options :)

**Comment by Totto16** [comment]
Yes of course, and its not in the docs since its hasnt been approved or accepted into the main branch

**Comment by FeeshDev** [comment]
I see embeds aren't working as well as the subcommand types, are you planning to work on those as well or will you wait on if / when it gets approved? if you even plan on doing the rest at all.

**Comment by Totto16** [comment]
I plan to do the rest also, but first I wanted to have some feedback, if the implementation is ok and if everything is fine with the existinf code

**Comment by Totto16** [comment]
Super if its going well, so I'll try to look at the types SUBCOMMAND and SUBCOMMAND_GROUP, the embed things isn't that complicated, in the moment you can pass embeds as {embeds:[embed]}, but thats a quick fix, something I also want to look into is the built in Permission Sytstem Discord offers for slash commands

**Comment by FeeshDev** [comment]
I'll try out the embed thing whenever I get the chance, I also wasn't aware they had built-in permissions which is great. Docs would definitely be helpful, however some of the stuff makes sense for people with experience or that have read discord slash command docs, so until approved I don't think it's a major thing, but then again I don't have much experience with this type of stuff, either way, good luck with it

**Comment by Totto16** [comment]
Thanks
